smack-integration-test/src/main/java/org/igniterealtime/smack/inttest/annotations/SpecificationReference.java

```diff
@@ -0,0 +1,55 @@
+/**
+ *
+ * Copyright 2024 Guus der Kinderen
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.igniterealtime.smack.inttest.annotations;
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * Reference to a specific part of a specification.
+ *
+ * @author Guus der Kinderen, guus@goodbytes.nl
+ */
+@Documented
+@Retention(RetentionPolicy.RUNTIME)
+@Target(ElementType.METHOD)
+public @interface SpecificationReference {
+
+    /**
+     * Unique identifier for a specification document, such as 'RFC 6120' or 'XEP-0485'.
+     *
+     * @return a document identifier
+     */
+    String document();
+
+    /**
+     * Unique identifier for a section (or paragraph) of the document referenced by {@link #document()}, such as '6.2.1'.
+     *
+     * @return a document section identifier
+     */
+    String section() default "";
```

Comment:
Is it possible to make the default `null` instead of the empty string (here and for 'quote')? That would allow us to reject annotations which contain the empty string as value.

Reply:
I don't think that's possible in Java.